homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,468 @@
+            vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
+            vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
+            vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
+            vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
+            vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+            vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(
+                ROUTE_MODE
+            ),
+            vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+            vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+        }
+    ),
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Dict[str, Union[str, bool]],
+    async_add_entities: Callable,
+    discovery_info: None = None,
+) -> None:
+    """Set up the HERE travel time platform."""
+    hass.data.setdefault(DATA_KEY, [])
+
+    app_id = config[CONF_APP_ID]
+    app_code = config[CONF_APP_CODE]
+    if config.get(CONF_ORIGIN_LATITUDE) is not None:
+        origin = ",".join(
+            [str(config[CONF_ORIGIN_LATITUDE]), str(config[CONF_ORIGIN_LONGITUDE])]
+        )
+    else:
+        origin = config[CONF_ORIGIN_ENTITY_ID]
+
+    if config.get(CONF_DESTINATION_LATITUDE) is not None:
+        destination = ",".join(
+            [
+                str(config[CONF_DESTINATION_LATITUDE]),
+                str(config[CONF_DESTINATION_LONGITUDE]),
+            ]
+        )
+    else:
+        destination = config[CONF_DESTINATION_ENTITY_ID]
+
+    travel_mode = config.get(CONF_MODE)
+    traffic_mode = config.get(CONF_TRAFFIC_MODE)
+    route_mode = config.get(CONF_ROUTE_MODE)
+    name = config.get(CONF_NAME)
+    units = config.get(CONF_UNIT_SYSTEM, hass.config.units.name)
+
+    here_data = HERETravelTimeData(
+        None, None, app_id, app_code, travel_mode, traffic_mode, route_mode, units
+    )
+
+    sensor = HERETravelTimeSensor(hass, name, origin, destination, here_data)
+
+    hass.data[DATA_KEY].append(sensor)
+
+    async_add_entities([sensor], True)
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        origin: str,
+        destination: str,
+        here_data: "HERETravelTimeData",
+    ) -> None:
+        """Initialize the sensor."""
+        self._hass = hass
+        self._name = name
+        self._here_data = here_data
+        self._unit_of_measurement = UNIT_OF_MEASUREMENT
+        self._origin_entity_id = None
+        self._destination_entity_id = None
+
+        # Check if location is a trackable entity
+        if origin.split(".", 1)[0] in TRACKABLE_DOMAINS:
```

Comment:
Will this work if origin is a coordinate?

Reply:
Yes it is tested by the tests.

Reply:
It's a bit weird that we split the coordinate and check if the integer part is in trackable domains. Unless we add a domain that is called like those integers, I guess it will keep working.